### PR TITLE
[4.x] Custom Middleware to contain auth features

### PR DESCRIPTION
**Pull request by ReeceM**
This PR should help improve the management of Auth for the package when accessing details.

## Description

The changes made is the creation of a new middleware called Authenticate. This is then added into the middleware config so it is loaded in the service provider.

It also adds the added functionality of having a method defined on the `User::class` model to have a `manageMailables` method where a person can define if that specific user has access to the Mailable editor.

It also has the change that it checks if the system is running in the console and then doesn't load up the environment check.

## Related Issue

#184  #114 

## Motivation and Context

This adds a central place to manage the authentication for the MailEclipse package. It also allows users to override the functionality if need be.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/Http/Middleware/Authenticate.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            && App::environment(config('maileclipse.allowed_environments', ['local']))
+        ) {
+            throw new UnauthorizedException('Environment not Allowed. Currently in '.App::environment());
+        }
+
+        if (method_exists($request->user(), 'manageMailables')) {
+            $request->user()->manageMailables() ?: throw new UnauthorizedException();
+        }
+
+        return $next($request);
```
Comment (ReeceM) [inline src/Http/Middleware/Authenticate.php:37]: Maybe this should just fail if the method isn't defined? Or allow it to pass if the environment is allowed?

**Comment by xicond** [comment]
Any update ?

**Comment by ReeceM** [comment]
@xicond is this something that you would like to use on the package?

**Comment by xicond** [comment]
@ReeceM I want to avoid error on artisan route:list when environment not exists in config

**Comment by ReeceM** [comment]
Hi There, I'll have a look tonight. Given some what the update for L10 as well I'll bundle the two together.

**Pull request closed without merging**